File: programmers/level1/02_하샤드_수.java
import java.util.*;

class Main {
    public static boolean solution(int x) {
        boolean answer = true;

        int digitTotal = 0;
        String[] arr = String.valueOf(x).split("");

        for (int i = 0; i < arr.length; i++) {
            digitTotal += Integer.parseInt(arr[i]);
        }

        if(x%digitTotal != 0){
            answer = false;
        }

        return answer;
    }

    public static void main(String[] args) {
        Main method = new Main();
        System.out.println(method.solution(12));
    }
}

File: programmers/level1/02_하샤드_수.py
def solution(x):
    answer = True
    num = 0

    new_list = list(str(x))

    for i in range(len(new_list)):
        num += int(new_list[i])

    if x % num != 0:
        answer = False

    return answer


s = solution(10)
print(s)
